homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -141,28 +207,62 @@ def device_state_attributes(self):
 
         return None
 
+    @property
+    def should_poll(self):
+        """Poll to retrieve latest data from Envoy endpoint."""
+        return True
+
+    @property
+    def available(self):
+        """Return if entity is available."""
+        return self.coordinator.last_update_success
+
+    async def async_added_to_hass(self):
+        """When entity is added to Home Assistant."""
+        self.async_on_remove(
+            self.coordinator.async_add_listener(self.async_write_ha_state)
+        )
+
+    @Throttle(MIN_SCAN_INTERVAL)
     async def async_update(self):
-        """Get the energy production data from the Enphase Envoy."""
+        """Update the energy production data."""
+
+        if self.coordinator.data is None:
+            _LOGGER.debug("No data found!")
+            return
+
         if self._type != "inverters":
-            _state = await getattr(self._envoy_reader, self._type)()
-            if isinstance(_state, int):
-                self._state = _state
+            if isinstance(self.coordinator.data.get(self._type), int):
+                self._state = self.coordinator.data.get(self._type)
+                _LOGGER.debug("Updating: %s - %s", self._type, self._state)
             else:
-                _LOGGER.error(_state)
-                self._state = None
+                _LOGGER.debug(
+                    "Sensor %s isInstance(int) was %s.  Returning None for state.",
```

Comment:
Can I suggest a slightly less technical message, such as "Data for sensor %s is not a number. ..."

Reply:
I figured for a debug message it didn't really need to be too readable.  Updated the message anyway 😄